### ISSUE TITLE
COCOeval vs MeanAveragePrecision

### ISSUE THREAD

**Issue by EmmaVanPuyenbr** [sovit-123/fasterrcnn-pytorch-training-pipeline#46]
In the eval.py you use AverageMeanPrecision from torchvision, in the train.py for evaluation you use the evaluation using pycocotools. When I use them both, I obtain different values for AP. which one do you recommend to use and why?

**Comment by sovit-123** [comment]
@EmmaVanPuyenbr 
I think I know the issue. When you use `eval.py` that uses Torchmetrics, it uses the best weights for the mAP calculation. 
You may be mistakenly comparing that the last epochs mAP from the validation loop. Please take a look at the log fine and find the best loop's mAP. I am sure that mAP and the Torchmetrics mAP will be the same.

**Comment by EmmaVanPuyenbr** [comment]
well I compare them after each epoch and even in the first epoch they are not the same? is it maybe also the mAP of the best iteration vs the last iteration?

**Comment by sovit-123** [comment]
Yes, that may be the case. Please check the terminal once for the best epoch mAP and then run `eval.py`.
Or else, you can get the best epoch from the CSV file as well.

**Comment by yun-ss97** [comment]
@EmmaVanPuyenbr hello, I'm having same issue that AP@0.5 with cocoeval and meanaverageprecision. Did you solve the probelm?

**Comment by yun-ss97** [comment]
I've found the solution for this issue. Add the code while loading model checkpoint!
```
model.transform.min_size = (args['img_size'], )
```
